protocol/activity_center_persistence.go

```diff
@@ -800,43 +818,21 @@ func (db sqlitePersistence) MarkActivityCenterNotificationsUnread(ids []types.He
 
 }
 
-func (db sqlitePersistence) buildActivityCenterNotificationsCountQuery(isAccepted bool, activityTypes []ActivityCenterType) *sql.Row {
-	var args []interface{}
-	var acceptedWhere string
-
-	if !isAccepted {
-		acceptedWhere = `AND NOT accepted`
-	}
-
-	var inTypeWhere string
-	if len(activityTypes) != 0 {
-		inVector := strings.Repeat("?, ", len(activityTypes)-1) + "?"
-		inTypeWhere = fmt.Sprintf(" AND notification_type IN (%s)", inVector)
-		for _, activityCenterType := range activityTypes {
-			args = append(args, activityCenterType)
-		}
-	}
-
-	query := fmt.Sprintf(`
-		SELECT COUNT(1) 
-		FROM activity_center_notifications 
-		WHERE NOT read AND NOT dismissed 
-		%s 
-		%s
-	`, acceptedWhere, inTypeWhere)
-
-	return db.db.QueryRow(query, args...)
-}
-
 func (db sqlitePersistence) UnreadActivityCenterNotificationsCount() (uint64, error) {
 	var count uint64
-	err := db.buildActivityCenterNotificationsCountQuery(false, []ActivityCenterType{}).Scan(&count)
+	err := db.buildActivityCenterNotificationsCountQuery(true, ActivityCenterQueryParamsReadUnread, []ActivityCenterType{}).Scan(&count)
 	return count, err
 }
 
 func (db sqlitePersistence) UnreadAndAcceptedActivityCenterNotificationsCount(activityTypes []ActivityCenterType) (uint64, error) {
 	var count uint64
-	err := db.buildActivityCenterNotificationsCountQuery(true, activityTypes).Scan(&count)
+	err := db.buildActivityCenterNotificationsCountQuery(false, ActivityCenterQueryParamsReadUnread, activityTypes).Scan(&count)
+	return count, err
+}
+
+func (db sqlitePersistence) ActivityCenterNotificationsCountBy(activityTypes []ActivityCenterType, readType ActivityCenterQueryParamsRead, accepted bool) (uint64, error) {
```

Comment:
It be better if we got a test for this public method. If in a hurry, maybe do it a follow-up PR?

Reply:
yep, we going to replace this method durning endpoints refactor